action/main.py

```diff
@@ -38,8 +44,25 @@
 base_cmd = [str(ENV_BIN / "black")]
 if BLACK_ARGS:
     # TODO: remove after a while since this is deprecated in favour of SRC + OPTIONS.
-    proc = run([*base_cmd, *shlex.split(BLACK_ARGS)])
+    proc = run([*base_cmd, *shlex.split(BLACK_ARGS)], stderr=PIPE)
 else:
-    proc = run([*base_cmd, *shlex.split(OPTIONS), *shlex.split(SRC)])
+    proc = run(
+        [*base_cmd, *shlex.split(OPTIONS), *shlex.split(SRC)],
+        stderr=PIPE
+    )
+# Re-emit stderr back to console so that action output is visible to pipeline
+# Do note, click will strip terminal control codes if the output is not TTY
+# and thus, this will not show colors anymore.
+print(proc.stderr, file=sys.stderr, flush=True)
```

Comment:
Im still wondering if i should proc.stderr should match sys.stdout.encoding ?

Reply:
and to give a bit of background, character encoding on github runners with windows as os, is reported to be `cp1252` while ubuntu and macos default to utf-8 .. enforcing utf-8 on subprocess.run() feels dirty if the output is going back to windows runner ..

Reply:
I guess, using sys.stdout.encoding is prob the better thing to do 
- Is there a `sys.stderr.encoding`? - I just know all our output will always be utf8 compliant ...

Reply:
Since I did miss this swallowing of the output, it's prob better to not encode @ subprocess (like you had it - sorry) and only decode for the regex and reprint the raw bytes. That might preserve the colors etc. too